### PR TITLE
Improvements, fixes and new features for the DWIN Enhanced UI

### PR DESCRIPTION
**Pull request by mriscoc**
### Description

Improvements, fixes and new features for the DWIN Enhanced UI

- Offset icon change to show mesh leveling status
- Reset extruder position when enter to Move menu
- New live end-stop diagnostic page
- Editable firmware retracts settings for Tune and filament settings menu
- Print Statistics page accessible from the Advanced Settings menu
- Reset printer draws the boot image
- Adds individual axes homing menu
- Adds probe deploy/stow to Probe Settings menu
- Updates lock screen
- Rebuilds main buttons to support text caption in other languages
- Increases probe offset limits to 60 mm
- Fix M303 PID variable update
- Fix Resume/Pause button update
- Fix redraw of print done
- Fix very large file name bug
- Fix bug in bed manual leveling

mesh-leveling-status  
The z-offset icon changes to indicate the active status of mesh leveling

endstops-diagnostic  
End-stop live status

tune-2  
Firmware retracts parameters

printer-stats  
Printer statistics

homing  
Individual axes homing

### Requirements

Ender 3v2 DWIN display

### Configurations

[Ender3v2.zip](https://github.com/MarlinFirmware/Marlin/files/7630797/Ender3v2.zip)

### CODE REVIEW

**Comment by Zwaubel** [comment]
@mriscoc 
Nice to see, that you continue to improve your menu implementation.
Do you by any chance plan to add support for 
- PID_EDIT_MENU
- PID_AUTOTUNE_MENU
- LEVEL_BED_CORNERS

menus in the future?

**Comment by mriscoc** [comment]
> 

Hi, those menus are already implemented in the DWIN Enhanced UI version.

**Comment by Zwaubel** [comment]
Oh, cool! 
Strangely the sanity checks are currently telling me that DWIN Enhanced UI would not support these features... there might be something wrong in my Repo though. I'll check that with a fresh clone of vanilla marlin.

**Comment by mriscoc** [comment]
> Oh, cool! Strangely the sanity checks are currently telling me that DWIN Enhanced UI would not support these features... there might be something wrong in my Repo though. I'll check that with a fresh clone of vanilla marlin.

Maybe because some defines are specific for MarlinUI.

**Comment by tome9111991** [comment]
> Oh, cool! Strangely the sanity checks are currently telling me that DWIN Enhanced UI would not support these features... there might be something wrong in my Repo though. I'll check that with a fresh clone of vanilla marlin.

You dont need to coment out.
The feature are allread  enabled

**Comment by thinkyhead** [comment]
@mriscoc — Is there anything new that you wanted to add to this before it is merged?

**Comment by mriscoc** [comment]
> @mriscoc — Is there anything new that you wanted to add to this before it is merged?

I think it's ok, thank you.

File: Marlin/src/lcd/language/language_en.h [hunk 1]
```diff
@@ -577,7 +586,8 @@ namespace Language_en {
   LSTR MSG_CASE_LIGHT_BRIGHTNESS          = _UxGT("Light Brightness");
   LSTR MSG_KILL_EXPECTED_PRINTER          = _UxGT("INCORRECT PRINTER");
 
-  #if LCD_WIDTH >= 20
+  #if (LCD_WIDTH >= 20 || HAS_DWIN_E3V2_BASIC)
```
Comment (thinkyhead) [inline Marlin/src/lcd/language/language_en.h]: Is `LCD_WIDTH` actually under 20 –or not defined– in the case of `DWIN_E3V2_BASIC`? Maybe we can fix that.

**Pull request merged**